test/libsolidity/semanticTests/various/selfdestruct_post_cancun_call_same_tx.sol

```diff
@@ -0,0 +1,51 @@
+contract C {
+    event Terminated();
+
+    constructor() payable {
+        assert(address(this).balance == 1 ether);
+        assert(address(msg.sender).balance == 0);
+        emit Terminated();
+        selfdestruct(payable(msg.sender));
+    }
+}
+
+contract D {
+    event Deployed(address);
+
+    C public c;
+
+    constructor() payable {}
+
+    function deploy() external payable {
+        c = new C{value: msg.value}();
+        emit Deployed(address(c));
+    }
+
+    function test() external payable {
+        (bool success, ) = address(this).call{value: 1 ether}(
+            abi.encodeCall(this.deploy, ())
+        );
+        assert(success);
```

Comment:
Same with all those low-level `CREATE2` tests. Doing it via `salt` is perfectly fine.

Reply:
> Is this meant to cover the case I mentioned in [#14785 (comment)](https://github.com/ethereum/solidity/pull/14785#discussion_r1457927986)?

No, the idea of this test was to check if the flag was being set correctly when a contract was created by a low-level call to create.

>  If so then it's not quite what I meant. I meant calling `C` _after_ it has already been created and destroyed. Here the creation happens during the call.
>
> So something along these lines:
> 
> ```
> C c = new C{value: 1 ether}();
> c.terminate();
> address(c).call(abi.encodeCall(C.f));
> ```
> 
> where `C.f()` is just an empty external function.
> 
> Also, the reason I mentioned a low-level call was that this way you could ignore a revert that should happen when you call a non-existent address. Asserting success kinda defeats the purpose :) If it doesn't revert we don't really need a low-level call here and `c.f()` would be fine.

If it doesn't revert, what would be the purpose of this test?

Reply:
> No, the idea of this test was to check if the flag was being set correctly when a contract was created by a low-level call to `create`.

But why would that change anything? `new` in the end will also generate bytecode with `create` or `create2`. It does not make much difference whether you used them in Yul yourself or if the codegen emitted them. It's not like the Solidity code is being interpreted at a high level here :)

> If it doesn't revert, what would be the purpose of this test?

The revert is not the main point here. The point was to call the contract after it was created because in your previous implementation that would overwrite the flag. In the current implementation it would no longer happen though, so it's irrelevant.